NuRadioReco/modules/channelGenericNoiseAdder.py

```diff
@@ -145,6 +160,27 @@ def bandlimited_noise(self, min_freq, max_freq, n_samples, sampling_rate, amplit
         elif type == 'rayleigh':
             fsigma = amplitude * sigscale / np.sqrt(2.)
             ampl[selection] = self.__random_generator.rayleigh(fsigma, nbinsactive)
+        elif type == "data-driven":
+            self.logger.warning("You have selected to generate data-driven noise NOTE THAT THE HARDWARE RESPONSE IS ALREADY INLUDED IN THIS NOISE and hence this module should be run AFTER incorporating hardware response.")
+            
+            if station_id is None or channel_id is None:
+                self.logger.error("When selecting data-driven noise, the station and channel ids should be passed to bandlimeted noise")
+                raise ValueError
+            
+            if station_id in [11]:
+                scale_parameter_path = self.scale_parameter_dir + "/" + f"thermal_noise_scale_parameters_s{station_id}_season23.json"
+            else:
+                raise NotImplementedError("Other station parameters are being generated")
+            
+            scale_parameter_dictionary = load_scale_parameters(scale_parameter_path)
+            scale_parameters = np.array(scale_parameter_dictionary["scale_parameters"])
```

Comment:
minor comment. the function `load_scale_parameters` could directly return `scale_parameters`?